src/mono/mono/mini/aot-compiler.c

```diff
@@ -15527,14 +15605,71 @@ emit_aot_image (MonoAotCompile *acfg)
 	return 0;
 }
 
+static int
+compile_assemblies_in_child (MonoAssembly **assemblies, int nassemblies, GPtrArray *runtime_args, const char *aot_options)
+{
+	/* Find --aot argument */
+	int aot_index = -1;
+	for (guint32 i = 1; i < runtime_args->len; ++i) {
+		const char *arg = (const char*)g_ptr_array_index (runtime_args, i);
+		if (strncmp (arg, "--aot=", strlen ("--aot=")) == 0) {
+			aot_index = i;
+			break;
+		}
+	}
+	g_assert (aot_index != -1);
+
+	GString *command;
+
+	command = g_string_new ("");
+
+	g_string_append_printf (command, "%s", (const char*)g_ptr_array_index (runtime_args, 0));
+
+	for (guint32 i = 1; i < runtime_args->len; ++i) {
+		const char *arg = (const char*)g_ptr_array_index (runtime_args, i);
+		if (strncmp (arg, "--response=", strlen ("--response=")) == 0)
+			/* Already expanded */
+			continue;
+		if (i != aot_index)
+			g_string_append_printf (command, " \"%s\"", arg);
+	}
+
+	/* Pass '_child' instead of 'compile-in-child' */
+	GPtrArray *aot_split_args = mono_aot_split_options (aot_options);
+	GString *new_aot_args = g_string_new ("");
+	for (guint32 i = 0; i < aot_split_args->len; ++i) {
+		const char *aot_arg = (const char*)g_ptr_array_index (aot_split_args, i);
+		if (i > 0)
+			g_string_append_printf (new_aot_args, ",");
+		if (!strcmp (aot_arg, "compile-in-child"))
+			g_string_append_printf (new_aot_args, "%s", "_child");
+		else
+			g_string_append_printf (new_aot_args, "%s", aot_arg);
+	}
+
+	g_string_append_printf (command, " \"--aot=%s\"", g_string_free (new_aot_args, FALSE));
+
+	for (int i = 0; i < nassemblies; ++i)
+		g_string_append_printf (command, " \"%s\"", assemblies [i]->image->name);
+
+	char *cmd = g_string_free (command, FALSE);
+	printf ("Executing: %s\n", cmd);
```

Comment:
I assume this can't be `aot_printf` because we'd try to write to the same log file as the parent? might be worth a comment. do we need to be careful not to use any other option that'd conflict with the parent?

Reply:
aot_printf requires a MonoAotCompile which we don't have here.